### PR TITLE
Add C# specific syntax highlighting

### PR DESCRIPTION
**Pull request by huipvandenende**
I love this theme but felt that it could be improved a bit for C# use. I changed some colors so it's easier to for example differentiate fields from functions. I also added a C# code example.

Old:
exampleOld

New:
exampleNew

### CODE REVIEW

File: themes/gruvbox-dark-hard.json [hunk 1]
```diff
@@ -479,6 +479,40 @@
...
+      }
+    },
+    {
+      "name": "C# This",
+      "scope": [
+        "keyword.other.this.cs",
+        "keyword.other.base.cs"
+      ],
+      "settings": {
+        "foreground": "#b16286"
```
Comment (jdinhify) [inline themes/gruvbox-dark-hard.json:513]: Could we try this with the light purple `#d3869b` and see how this goes?
If we try & decide to go with the light purple for dark variants, perhaps we'll need to look at dark purple `#8f3f71` for the light variants.

**Review by jdinhify — COMMENTED** [review]
Thanks @huipvandenende for this PR, and apologies for the late review.

Just a comment to see whether we should use dark purple instead for `C# this`. Please try and see which one you prefer

**Comment by jdinhify** [comment]
Also please add yourself to the contributors section in `package.json` & `README.md` 🙏

**Comment by huipvandenende** [comment]
I have updated the colors so "this" is consistent with the other purple elements. Thanks for pointing that out.

(Excuse the fat fingers)

**Review by jdinhify — APPROVED** [review]
Thanks @huipvandenende , I'll merge this & deploy soon

**Pull request merged**